csharp-client-reference/MainWindow.xaml.cs

```diff
@@ -309,14 +314,19 @@ private async void Register_Click(object sender, RoutedEventArgs e)
             //TODO better error state handing, when register failed due to no connection, it is not well though right now
             //TODO also when error flow when register e.g. same name
             var registrationCommandDto = new RegistrationCommandDTO {Name = RegName.Text };
+            if (_requests != null)
```

Comment:
Request is null checked to avoid calling register twice.

Reply:
This is a temporary fix, and here I try not to adding more message type for `OASISQueryDTO` just to keep minimal change on API for now. Also, I was struggling to find out a better way separating connection and register stuff.

Reply:
So what does:

value of `registered` | value of `requests` | meaning
--- | --- | ---
`false` | `null` | node is not registered
`false` | `Channel@1234` | ???
`true` | `null` | ???
`true` | `Channel@3456` | node is registered and stream is open

can you fill in the ???'s?

Reply:
Ah, we don't need that boolean anymore, going to remove this. If the `_request` is null, it is consider unregistered.